### DISCUSSION TITLE
Possible Integration with ligolo-ng

### DISCUSSION THREAD

**Discussion by 3m0W33D** [BishopFox/sliver#1369] [Ideas]
I recently saw under the issues that there are quite a few issues with the existing socks5 proxy. Coincidentally, I saw this tool https://github.com/nicocha30/ligolo-ng, that provides a pretty cool way to pivot by creating a VPN like tunnel and it also supports socks5 proxies within it. Is it be possible to include this as a armoury plugin or even be integrated into sliver? I believe this would be a pretty cool addition.

**Comment by rkervella** [comment]
First party integration would require a lot of work, but you can likely still use `sideload` to run that or modify it a bit to be loaded as an extension for Windows targets (extensions are currently not supported on other systems).

**Reply by 3m0W33D** [reply]
Ill definitely try to `sideload` to load it into linux/windows! But I think it would have been a good to mention this idea as many seems to have issues with the socks5 proxy. Thanks for taking the time to answer this idea's feasibility!

**Comment by n3rada** [comment]
It's true that it's in go as well. It would be incredible to have a compatible operation on `sliver`. You could just start your `proxy` on your attack machine and use the implants to launch from elsewhere!!

**Reply by rkervella** [reply]
Make a PR and we'll have a look :)

**Reply by n3rada** [reply]
Unfortunately learning `go` is far away on my to-do list, it was more to show my enthusiasm for the author's idea 👀.
